server_management/management/commands/_core.py

```diff
@@ -135,19 +136,62 @@ def get_remote(remote):
     return remote_prompt
 
 
-def title_print(title, state=''):
+def prompt(text, default=None, validate=None):
+    if default:
+        text = text.strip()
+        if text[-1] == ':' and len(text) > 1:
+            text = text[0:-1]
+        text = f'{text} [{default}]: '
+
+    while 1:
+        user_input = input(text)
+
+        if not user_input and default is not None:
+            return default
+
+        if (user_input and validate(user_input)) or (user_input and default is None):
```

Comment:
`validate` is permitted to be `None` here - will this not throw an exception on any user input where `validate` is not supplied?

Reply:
The second part was meant to be `validate is None` but it should be first anyway